libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationPresenter.java

```diff
@@ -69,7 +69,7 @@ void onNavigationLocationUpdate(Location location) {
   }
 
   void onWayNameChanged(@NonNull String wayName) {
-    if (TextUtils.isEmpty(wayName)) {
+    if (TextUtils.isEmpty(wayName) || view.isSummaryBottomSheetHidden()) {
```

Comment:
Right now, I believe the feature filter task is executed anyways (although we're not showing the way name for the cases in which shouldn't be shown). Is there any possibility to achieve this but also not execute `FeatureFilterTask` at all so it's not calculated in the cases in which the way name should be not visible?